pkgs/test_core/lib/src/runner/parse_metadata.dart

```diff
@@ -24,8 +24,11 @@ import '../util/dart.dart';
 ///
 /// Throws an [AnalysisError] if parsing fails or a [FormatException] if the
 /// test annotations are incorrect.
-Metadata parseMetadata(String path, Set<String> platformVariables) =>
-    _Parser(path, platformVariables).parse();
+///
+/// If [contents] is `null` then it will be read as a [File] using [path].
+Metadata parseMetadata(String path, Set<String> platformVariables,
```

Comment:
I would make this always take `content` and update the calling locations to read the file themselves.

Reply:
Did this slip?

Is it still worth doing? We haven't published yet...

Reply:
Oh weird - somehow I didn't see this comment when I was merging. I will go ahead and send out another PR for it.